functions/src/vote.ts

```diff
@@ -1,32 +1,34 @@
 import { firestore } from "firebase-admin";
 import type { DocumentReference, Transaction } from "firebase-admin/firestore";
-import { GAME_STATE_DURATIONS_DEFAULT, Lobby, Player } from "./firestore-types/lobby";
+import { GAME_STATE_DURATIONS_DEFAULT, Lobby } from "./firestore-types/lobby";
 
 export function findVoteOff(lobbyData: Lobby, lobbyDocRef: DocumentReference<Lobby>, transaction: Transaction) {
-  const { players, uids } = lobbyData;
+  const { players } = lobbyData;
   let votedOff: string | undefined = undefined;
 
   // find the player with the most votes
-  const [most, secondMost]: Player[] = JSON.parse(JSON.stringify(players)).sort((a: Player, b: Player) => {
-    return b.votes - a.votes;
-  });
-  // if they're not equal and have less votes than skip then most is voted off
-  if (most.votes != secondMost.votes && most.votes > lobbyData.skipVote) {
-    // find index of most in the players array
-    let mostIndex: number | undefined;
-    for (let i = 0; i < players.length; i++) {
-      const player = players[i];
-      if (player.votes == most.votes) {
-        mostIndex = i;
-      }
+  let most: string | null = null;
+  let mostValue = 0;
+
+  for (const uid in lobbyData.players) {
+    if (players[uid].votes > mostValue || players[uid].votes > lobbyData.skipVote) {
+      most = uid;
+      mostValue = players[uid].votes;
+    } else if (players[uid].votes == mostValue || players[uid].votes == lobbyData.skipVote) {
+      most = null;
     }
```

Comment:
I think you don't want to have the || skipVote here because it means that any player that has more votes than skip is voted off.

Why not just `let mostValue = lobbyData.skipVote` ?